x/mongo/driver/operation.go

```diff
@@ -86,6 +86,15 @@ type finishedInformation struct {
 	serverID  *primitive.ObjectID
 }
 
+// ResponseInfo contains the context required to parse a server response.
+type ResponseInfo struct {
+	ServerResponse        bsoncore.Document
+	Server                Server
+	Connection            Connection
+	ConnectionDescription description.Server
```

Comment:
Is this field redundant? Can you not access a `description.Server` through `Connection.Description()`?

Reply:
You can. However, I'm hesitant to remove it because that would require a lot more changes than we have now. Also, accessing `Connection.Description()` internally requires a mutex, so I feel like storing the cached description here is still valuable. Thoughts?

Reply:
Ah great point about the mutex 🔒 . Seems fine to me!